drivers/charger/charger_max20335.c

```diff
@@ -234,26 +388,160 @@ static int max20335_get_prop(const struct device *dev, charger_prop_t prop,
 static int max20335_set_prop(const struct device *dev, charger_prop_t prop,
 			     const union charger_propval *val)
 {
+	struct charger_max20335_data *data = dev->data;
+
 	switch (prop) {
 	case CHARGER_PROP_CONSTANT_CHARGE_CURRENT_UA:
-		return max20335_set_constant_charge_current(dev,
-							    val->const_charge_current_ua);
+		data->charge_current_ua = val->const_charge_current_ua;
```

Comment:
These values should be set only if the corresponding function call succeeded to avoid a state where the stored values do not match the the charger has actually been set to.

Reply:
Hi @GRobertZieba 

To make sure that we are on the same page please have a look on a picture below.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/124128327/6ac4affd-bc54-491f-b7c1-5a32e8978e24)

This is how the hardware and my code behaves.
I have a few questions related to your suggestion:

* What  value `max20335_get_prop()` should give to the caller if the interrupt triggering the config did not occur i.e until power supply is connected?
* Would you like me to implement something like shadow variables whoch buffer requested parameters values and additionaly parallel set of variables that will be updated when configuration takes place and succeeds?

Reply:
Hello @LukaszMadejGrinn, is the reset behavior configurable?

I'm hesitant to drop support for those that may be polling this device rather than relying on the INT line. However, if this reset behavior is fixed in the HW, it sounds like the change you are proposing is a "must-have". I'm not super familiar with this series of charger PMICs.

Reply:
Hey @LukaszMadejGrinn,

`max20335_set_prop` doesn't just buffer the values, but sets them through `max20335_set_constant_charge_voltage` and `max20335_set_constant_charge_voltage`. But your diagram only shows the registers getting configured after an interrupt. Is that the intended behavior? Or should the configuration happen both during `max20335_set_prop` and after an interrupt?

My concern is that if `max20335_set_prop` gets called and either of those functions fails then that value gets buffered and would get applied after the next interrupt, instead of maintaining the previous value.

Reply:
Hi @rriveramcrus

I understand your hesitation but unfortunately it is mandatory. More or less all registers that configures operation of the charger are reset to default values upon connection of a power supply. Therefore writing any value to the charger before applying the supply does not really make sense. Please see snippets from DS below and an asterix mark:

_Register is reset to default value upon CHGIN rising edge_

![image](https://github.com/zephyrproject-rtos/zephyr/assets/124128327/51234206-cfb3-4af5-ae91-cf94d68bb797)

![image](https://github.com/zephyrproject-rtos/zephyr/assets/124128327/510c6f62-4899-47b9-bfae-2b39ecaebfbc)

Reply:
Hey @GRobertZieba 

> `max20335_set_prop` doesn't just buffer the values, but sets them through `max20335_set_constant_charge_voltage` and `max20335_set_constant_charge_voltage`. But your diagram only shows the registers getting configured after an interrupt. Is that the intended behavior? Or should the configuration happen both during `max20335_set_prop` and after an interrupt?
> 
> My concern is that if `max20335_set_prop` gets called and either of those functions fails then that value gets buffered and would get applied after the next interrupt, instead of maintaining the previous value.

I think "buffering" a value was not used correctly by me in this context and made a confusion. Actually the prop is set indeed with `max20335_set_prop` and it is as you said i.e. it can be obtained with `max20335_get_prop` and this obtained value does not need to reflect value stored in the charger fir example due to some error occurrence.

There are some solutions for that but I'm wondering what is the expected behavior.
One of possible implementation is the following:

1. Requested parameters values with `max20335_set_prop` is stored by the driver in RAM.
2. Until the value is written to a charger the `max20335_get_prop` returns `-EAGAIN`.
3. If the value is written to a charger the `max20335_get_prop` returns written value.
4. If the value is written to a charger the `max20335_get_prop` returns `-EIO`.


On the other hand there are further changes prepared by me that will be included in further MR once some other are accepted. Please have a look on this branch:
https://github.com/LukaszMadejGrinn/zephyr/commits/charger-max20335-make-cfg-complete/

The parameters we are discussing right now are not used properly and this needs to be fixed as well. However this is not a scope of this PR thus not included yet. To sum up, we can try to achieve a perfect implementation right now but the code will be removed by the end of the day. So it this worth efforts? In my opinion not, especially due to a fact that I have limited time for upstreaming those modifications ;)

My proposition here is to keep this implementation as is and introduce improvement on this area (if still necessary) within the next PR I'm gonna create.

What do you think about this?

Reply:
@GRobertZieba Fixed